project_milestone_enhanced/models/project_task.py

```diff
@@ -0,0 +1,52 @@
+# © 2022 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models
+
+
+class ProjectTask(models.Model):
+
+    _inherit = "project.task"
+
+
+    @api.onchange('project_id')
+    def _onchange_project(self):
+        result = super(ProjectTask, self)._onchange_project()
+        result = result or {}
+        if self.project_id.use_milestones :
+            self.milestone_id = False
+        if self.project_id.sale_order_id:
```

Comment:
Il faut ajouter la condition "and self.project_id.use_milestones" 
Sinon le onchange va vider le champ sale_line_id pour les autres types de projet
(Voir **Exception** dans la SFD)